python/Ganga/Core/InternalServices/Coordinator.py

```diff
@@ -234,10 +180,10 @@ def enableInternalServices():
     Repository_runtime.bootstrap()
 
     # make sure all required credentials are valid
-    missing_cred = getMissingCredentials()
-    if missing_cred:
-        log.error("The following credentials are still required: %s."
-                  "Make sure you renew them before reactivating this session" % ','.join(missing_cred))
+    invalid_afs = [afsToken for afsToken in credential_store.get_all_matching_type(AfsToken) if not afsToken.is_valid()]
+
+    if invalid_afs:
+        log.error('No valid AFS token was found. Please re-authorise before reactivating this session.')
```

Comment:
If the AFS token expires we get horrible errors and nothing makes sense and eventually everything bails out because we can't even log to disk.
I would just be worried about submitting jobs to a remote backend and then losing all track of them here because Ganga isn't flushing until after the jobs have been submitted. It's a bad situation to be in.